### ISSUE TITLE
Before calculate loss, why apply softmax in validate_simdr but not in train_simdr?

### ISSUE THREAD

**Issue by shizidushu** [leeyegy/SimDR#13]
Why apply softmax in `validate_simdr`:
https://github.com/leeyegy/SimDR/blob/67216117f08d2ccfb65324859758aa007976831a/lib/core/function.py#L324-L328

And not apply softmax in `train_simdr`?
https://github.com/leeyegy/SimDR/blob/67216117f08d2ccfb65324859758aa007976831a/lib/core/function.py#L249-L255

I guess softmax operation is not needed in the validate phase? As there is an [`LogSoftmax` operation ](https://github.com/leeyegy/SimDR/blob/67216117f08d2ccfb65324859758aa007976831a/lib/core/loss.py#L30) to the output when calculating loss.

**Comment by leeyegy** [comment]
Thanks for your attention.
When training, as the same as typical classification task, softmax operation is actually in loss function.
When validating, softmax is used to generate a probability distribution, which is useful for the following operations (e.g., for the visible threshold).
